servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/SingleConcatWithPublisher.java

```diff
@@ -152,7 +152,12 @@ public void cancel() {
         }
 
         private void emitSingleSuccessToTarget(@Nullable final T result) {
-            target.onNext(result);
+            try {
+                target.onNext(result);
+            } catch (Throwable cause) {
+                target.onError(cause);
```

Comment:
Do we need to use `SubscriberUtils#safeOnError` here?

Reply:
upstream should handle exceptions being thrown and won't propagate a duplicate terminal.